lib/core/refresh.js

```diff
@@ -0,0 +1,123 @@
+import ExpiredAuthSessionError from './includes/ExpiredAuthSessionError'
+
+export class RefreshController {
+  constructor (scheme) {
+    this.scheme = scheme
+    this.$auth = scheme.$auth
+    this.refreshPromise = null
+    this.refreshInterval = undefined
+  }
+
+  // Returns a promise which is resolved when refresh is completed
+  // Call this function when you intercept a request with an expired token.
+  // Multiple requests will be queued until the first has completed token refresh.
+  handleRefresh () {
+    // Another request has started refreshing the token, wait for it to complete
+    if (this.refreshPromise) {
+      return this.refreshPromise
+    }
+
+    return this._doRefresh()
+  }
+
+  _doRefresh () {
+    this.refreshPromise = new Promise(resolve => {
+      this.scheme.refreshToken().then(response => {
+        this.refreshPromise = null
+        resolve(response)
+      })
+    })
+
+    return this.refreshPromise
+  }
+
+  initializeAutomaticRefresh () {
```

Comment:
Small thing: Inconsistency `Auto` vs `Atuomatic`

Reply:
This method was removed in commit https://github.com/nuxt-community/auth-module/pull/548/commits/f90fe79ce830e63500cc5bdaf515e7d9cf5c63f3